src/java_bytecode/java_object_factory.cpp

```diff
@@ -1028,6 +1030,27 @@ void java_object_factoryt::gen_nondet_struct_init(
         substruct_in_place);
     }
   }
+
+  // If <class_identifier>.cproverValidate() can be found in the
+  // symbol table, we add a call:
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  // expr.cproverValidate();
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  const irep_idt validate_method_name =
+    "java::" + as_string(class_identifier) + ".cproverNondetInitialize:()V";
+
+  if(const auto func = symbol_table.lookup(validate_method_name))
+  {
+    const code_typet &type = to_code_type(func->type);
+    INVARIANT(
```

Comment:
I'd argue for warn-and-ignore rather than crash, since this is user-provided code, not an internal invariant

Reply:
There is no `messaget` object available here to log a warning. Is throwing ok?

Reply:
I'd pass one in (from `java_bytecode_languaget`), I think warning is better than killing the whole run.

Reply:
I tried something like that, but ended up having to pass it through an endless number of functions, which would probably end up tripling the size of this PR. 
Instead I found making this work even if the method is static much easier.
If the user needed a non-static method then they should notice it at compilation.
For the check on the number of arguments this seems unnecessary since the signature already gives the information.